pkg/catalogv2/git/download.go

```diff
@@ -57,47 +58,26 @@ func Head(secret *corev1.Secret, namespace, name, gitURL, branch string, insecur
 	return commit, nil
 }
 
-// Update updates git repo if remote sha has changed
```

Comment:
The previous code comment was a bit redundant, but I think there is enough going on in the function to warrant a small description.
It doesn't just update, it also skips the update if in bundled mode and the dir has a certain prefix. If find it especially interesting when this returns an error.
E.g. when working on the state dir, but not in "bundled" mode, it will return an error only for `couldn't find remote ref` and `Could not find remote branch` different.